src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,260 @@
+/**
+ * @file methods/kernel_svm/kernel_svm_impl.hpp
+ * @author Himanshu Pathak
+ *
+ * Implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "kernel_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const double C,
+    const bool fitIntercept,
+    const size_t max_iter,
+    const double tol,
+    const KernelType& kernel) :
+    C(C),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  b = 0;
+  Train(data, labels, max_iter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const size_t inputSize,
+    const double C,
+    const bool fitIntercept,
+    const KernelType& kernel) :
+    inputSize(inputSize),
+    C(C),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  b = 0;
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVM<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t max_iter,
+    const double tol)
+{
+  arma::Row<int> label(data.n_cols);
+  for (size_t i = 0; i< data.n_cols; i++)
+  {
+    if (labels(i) == 0)
+      label(i) = -1;
+    else
+      label(i) = 1;
+  }
+
+  alpha = arma::zeros(data.n_cols);
+  size_t count = 0;
+  arma::vec E = arma::zeros(data.n_cols, 1);
+  double eta = 0;
+  double L = 0;
+  double H = 0;
```

Comment:
can you explain what these values are for? can you leave more comments to explain the purpose of these?

Reply:
@himanshupathak21061998 can you please address this?